autogpt_platform/backend/backend/executor/database.py

```diff
@@ -50,17 +54,11 @@ def send_execution_update(self, execution_result: ExecutionResult):
     def exposed_run_and_wait(
         f: Callable[P, Coroutine[None, None, R]]
     ) -> Callable[Concatenate[object, P], R]:
-        @expose
-        @wraps(f)
-        def wrapper(self, *args: P.args, **kwargs: P.kwargs) -> R:
-            coroutine = f(*args, **kwargs)
-            res = self.run_and_wait(coroutine)
-            return res
-
-        # Register serializers for annotations on bare function
-        register_pydantic_serializers(f)
-
-        return wrapper
+        # TODO:
+        #  This function lies about its return type to make the DynamicClient
+        #  call the function synchronously, fix this when DynamicClient can choose
+        #  to call a function synchronously or asynchronously.
+        return expose(cast(Callable[Concatenate[object, P], R], f))
```

Comment:
nit: if the return type is known to be incorrect, don't use `cast` but `# type: ignore`:
```suggestion
        # TODO:
        #  This function lies about its return type to make the DynamicClient
        #  call the function synchronously, fix this when DynamicClient can choose
        #  to call a function synchronously or asynchronously.
        return expose(f)  # type: ignore
```